imapclient/client_test.go

```diff
@@ -16,11 +18,34 @@ const (
 	testPassword = "test-password"
 )
 
+type FileLiteralReader struct {
+	*os.File
+}
+
+func (r FileLiteralReader) Size() int64 {
+	fInfo, err := r.Stat()
+	if err != nil {
+		panic(fmt.Errorf("Stat: %w", err))
+	}
+	return fInfo.Size()
+}
+
 func newClientServerPair(t *testing.T, initialState imap.ConnState) (*imapclient.Client, io.Closer) {
 	memServer := imapmemserver.New()
 
 	user := imapmemserver.NewUser(testUsername, testPassword)
 	user.Create("INBOX", nil)
+
+	emlFile, err := os.OpenFile("../testdata/sample.eml", os.O_RDONLY, 0)
```

Comment:
Can we maybe use `strings.NewReader` with a multi-line string literal? That should reduce the boilerplate a bit. Also the test message doesn't need to be very complicated, just one or two header and a 1-line body should be enough.